lib/astarte_core/mapping.ex

```diff
@@ -100,6 +100,7 @@ defmodule Astarte.Core.Mapping do
     |> validate_length(:description, max: 1000)
     |> validate_length(:doc, max: 100_000)
     |> normalize_fields()
+    |> check_coherence_database_retention_policy_and_database_retention_ttl()
```

Comment:
Move this above `normalize_fields` so that we perform all validation first and then possibly transform data